### DISCUSSION TITLE
Localization contribution docs update

### DISCUSSION THREAD

**Discussion by divya-mohan0209** [kubernetes/website#42409] [General]
In line with the membership requirements of the Kubernetes community, our [localization docs](https://kubernetes.io/docs/contribute/localization/) in the contributor guide need an update.

But as SIG Docs, we'd like to take this opportunity to crowdsource ideas from localization members & the wider SIG Docs contributors to make the docs more comprehensive & up-to-date.

This discussion item will be open for a period of 14 days, after which we will be opening up an issue to update the respective section/s in the docs.

cc: @natalisucks @reylejano

**Comment by seokho-son** [comment]
FYI: https://github.com/kubernetes/community/blob/master/community-membership.md#requirements

**Comment by a-mccarthy** [comment]
Thanks for creating this Divya! 

Reading localization contributing guide, it feels a little overwhelming because it covers everything from how to be a casual contributor to launching a brand new localization. Maybe we could consider breaking it into a few different pages with a smaller target audience?

**Comment by a-mccarthy** [comment]
Some other ideas:

* Include some content or resources on how to build a community and why it is important? This might be more of a blog, but I think it's important to help folks to realize the importance of finding and building community for the long term stability of the localization. 
* Diagrams for the branching strategy section or a link yo review the diagrams on the Contributing page
* Explain more about maintain docs from older releases, see details in slack thread: https://kubernetes.slack.com/archives/C0191RDKHU1/p1692198797857159

**Comment by seokho-son** [comment]
Maybe it is beneficial to add a section to guide how to manage `_redirect` URLs for localized pages.
(but the guide can be changed based on a discussions in https://github.com/kubernetes/website/pull/42629)

**Comment by seokho-son** [comment]
FYI: Current headers in [Localizing Kubernetes documentation](https://kubernetes.io/docs/contribute/localization/)

## 1 Contribute to an existing localization
- Find your two-letter language code
- Fork and clone the repo
- Suggest changes

## 2 Start a new localization
- Find community
- Join the Kubernetes GitHub organization
- Add your localization team in GitHub
- Configure the workflow
- Modify the site configuration
- Add a new localization directory
- Localize the community code of conduct
- Setting up the OWNERS files
- Open a pull request
- Add a localized README file
- Launching your new localization

## 3 Translating content
- Minimum required content
- Translating SVG images
- Source files
- Site strings in i18n
- Language-specific localization guide
- Language-specific Zoom meetings

## 4 Branching strategy

## 5 Upstream contributions

**Reply by seokho-son** [reply]
My draft suggestion to reorganize the content.

- Localizing Kubernetes documentation (includes 3 sub-pages)
  - Start a new localization
    - Find community
    - Join the Kubernetes GitHub organization
    - Add your localization team in GitHub
    - Configure the workflow
    - Modify the site configuration
    - Add a new localization directory
    - Localize the community code of conduct
    - Setting up the OWNERS files
    - **Site strings in i18n**
    - Open a pull request
    - Add a localized README file
    - **Minimum required content** (add link to `Localize content`)
    - Launching your new localization
  - Contribute to an existing localization
    - Fork and clone the repo
    - Find your two-letter language code
    - Suggest changes 
    - **Localize content**
      - ...
      - 
      - **Localize SVG images**
    - Upstream contributions
  - **Maintain an existing localization** 
    - (Importance of long term stability of the localization)
    - Language-specific localization guide
    - Language-specific Zoom meetings
    - Source files
    - **Branching strategy**

**Reply by natalisucks** [reply]
I have a slight preference to keep `Contribute to an existing localization` before our steps to starting a new one, given the workload of starting new, but otherwise I like the rest of your suggestions for reorganizing the list @seokho-son 👍

**Reply by a-mccarthy** [reply]
I like this organization @seokho-son !  I also agree with Natali, I think it makes sense to have "Contribute to an existing localization" first because I think thats what the most common use of this section will be. Some other ideas for the "Maintain an existing localization" page that it should maybe also cover some recommended for responsibilities of localization maintainers and when to add/remove new members in the owners files

**Comment by a-mccarthy** [comment]
closing this discussion and moving work into an issue: https://github.com/kubernetes/website/issues/43116 If you have further comments or would like to check the status, please see the issue.
